hazelcast-spring/src/test/resources/com/hazelcast/spring/advancedNetworkConfig-applicationContext-hazelcast.xml

```diff
@@ -100,7 +100,7 @@
                         <hz:property name="maxEndpoints">2</hz:property>
                     </hz:properties>
                 </hz:wan-publisher>
-                <hz:wan-publisher group-name="ankara" class-name="com.hazelcast.enterprise.wan.replication.WanBatchReplication">
+                <hz:wan-publisher group-name="ankara" class-name="com.hazelcast.enterprise.wan.impl.replication.WanBatchReplication">
```

Comment:
Referencing an impl class doesn't seem ideal. If we never expect a publisher implemented by user, hiding it completely can be a good option.

Reply:
I think even if the user can implement it, this class can be resolved internally without forcing the user to set it. This class is not even designed for extension, so :+1: for making it private.

Reply:
Yes, it was one of the things that I was a bit concerned about but not too much. For one, it definitely is internal in the sense that we don't support the user to instantiating it using `new com.hazelcast.enterprise.wan.impl.replication.WanBatchReplication();` and using it somewhere in their code. It's our internal implementation and it's satisfactory for most users using WAN.

The problem here may be that we require the fully qualified class name. I was thinking that we might add some aliases like `WanBatchReplication` so it's easier to configure and easier on the eyes. The other idea was simply to make it the default as most users will use it anyway. So this could work then:
```
<hz:wan-publisher group-name="ankara">
...
</hz:wan-publisher>
```

But either of these ideas would be part of a subsequent PR, not this one. WDYT?

Reply:
One more thought. Maybe instead of the FQCN of the endpoint implementation, we should accept a factory's FQCN. So that additional dependencies can be injected in the way the user likes.